Topics/Development_Process/Trunk_Development.md

```diff
@@ -0,0 +1,5 @@
+### Trunk-Based Development
+
+Knowing how to manage the creation of large new features in your repository is an important skill, especially when working in large organizations or on projects with complete CI systems. It can be tempting to have large feature branches with hundreds of new lines of code and it can seem daunting to implement a large feature in small stages. Trunk-based development is a system to handle large changes like this. It focuses on creating small PR and using tools such as feature flags. The following article from Atlassian does a great job discussing all the benefits of trunk-based development (i.e. impact on automated testing, having fewer active branches, and more): **[Atlassian Article](https://www.atlassian.com/continuous-delivery/continuous-integration/trunk-based-development)**
+
+Instead of using feature flags, one can also use the "branch by abstraction" method found in [this article](https://trunkbaseddevelopment.com/branch-by-abstraction/). This is similar to referencing an interface instead of a class and much like dependency inversion, can both increase or reduce the complexity of your code.
```

Comment:
Since feature flags are quite a useful tool and we're even using them in our own partner project, would you be able to write a short section in between your first and second paragraph describing how feature flags can be used/implemented in trunk-based development? Maybe a concrete example of how they help us break down the implementation of a "post on social media" feature into digestible batches and what type of flags may be derived in that example (doesn't have to be social media, just the first thing that came to mind).

And then the transition into the next paragraph `Instead of using feature flags,` I think will have a bit more meaning as now they can contrast the different methods of breaking down a large feature.

Reply:
Great idea! I will ad a more concrete example of how one might use them :+1: